pydrive2/auth.py

```diff
@@ -139,6 +141,29 @@ def _decorated(self, *args, **kwargs):
     return _decorated
 
 
+class MemoryStorage(ClientStorage):
+    def __init__(self, creds_json=None):
+        self._json = creds_json
+        super().__init__()
+
+    def locked_get(self):
+        if not self._json:
+            return None
+
+        try:
+            creds = Credentials.new_from_json(self._json)
+            creds.set_store(self)
+            return creds
+        except ValueError:
+            return None
+
+    def locked_put(self, credentials):
+        self._json = credentials.to_json()
+
+    def locked_delete(self):
+        self._json = None
```

Comment:
Ah, obviously someone contributed similar thing in https://github.com/googleapis/oauth2client/blob/50d20532a748f18e53f7d24ccbe6647132c979a9/oauth2client/contrib/dictionary_storage.py Will use it...